Frameworks/Foundation/NSOperation.mm

```diff
@@ -277,35 +244,35 @@ - (void)main {
  @Status Interoperable
 */
 - (void)waitUntilFinished {
-    pthread_mutex_lock(&priv->finishLock);
+    pthread_mutex_lock(&_finishLock);
     while (![self isFinished]) {
-        pthread_cond_wait(&priv->finishCondition, &priv->finishLock);
+        pthread_cond_wait(&_finishCondition, &_finishLock);
     }
-    pthread_mutex_unlock(&priv->finishLock);
+
+    pthread_mutex_unlock(&_finishLock);
 }
 
 /**
  @Status Interoperable
 */
-- (id)dependencies {
-    return priv->dependencies;
+- (NSArray*)dependencies {
+    pthread_mutex_lock(&_dependenciesLock);
+    NSArray* copy = [[_dependencies copy] autorelease];
+    pthread_mutex_unlock(&_dependenciesLock);
+    return copy;
 }
 
 /**
  @Status Interoperable
 */
 - (void)dealloc {
-    assert(!priv->completionBlock);
-    delete priv;
+    [self removeObserver:self forKeyPath:@"isFinished" context:NULL];
+    pthread_cond_destroy(&_finishCondition);
+    pthread_mutex_destroy(&_finishLock);
+    pthread_mutex_destroy(&_dependenciesLock);
+    pthread_mutex_destroy(&_completionBlockLock);
+    [_completionBlock release];
```

Comment:
`_completionBlock` can be a `StrongId<void(^)()>` if you want to dispense with the formalities.